src/api/Controllers/ImportController.cs

```diff
@@ -236,12 +156,231 @@ public async Task<ActionResult<int>> UploadFileAsync(int workgroupId, IFormFile
+                foreach (var fileToProcess in filesToProcess)
+                {
+                    var fileName = $"{fileToProcess.File.NameUuid}_{fileToProcess.File.Name}";
+                    var attachment = zipArchive.Entries.FirstOrDefault(e => e.FullName == fileName);
+
+                    using var memoryStream = new MemoryStream();
+                    FormFile formFile = await CreateFormFileFromAttachmentAsync(fileToProcess, attachment, memoryStream).ConfigureAwait(false);
+
+                    // Remove original file information from borehole object
+                    borehole.value.BoreholeFiles.Remove(fileToProcess);
+
+                    await UploadFormFileAsync(borehole.value, borehole.index, fileName, formFile).ConfigureAwait(false);
+                }
+            }
+        }
+    }
+
+    private async Task UploadFormFileAsync(BoreholeImport borehole, int boreholeIndex, string fileName, FormFile formFile)
+    {
+        try
+        {
+            await boreholeFileCloudService.UploadFileAndLinkToBorehole(formFile, borehole.Id).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "An error occurred while uploading the file: {FileName}", formFile.FileName);
+            AddValidationErrorToModelState(boreholeIndex, string.Format(CultureInfo.InvariantCulture, $"An error occurred while uploading the file: <{fileName}>", "upload"), ValidationErrorType.Attachment);
+        }
+    }
+
+    private static async Task<FormFile> CreateFormFileFromAttachmentAsync(BoreholeFile? boreholeFile, ZipArchiveEntry? attachment, MemoryStream memoryStream)
+    {
+        using (var entryStream = attachment.Open())
+        {
+            await entryStream.CopyToAsync(memoryStream).ConfigureAwait(false);
+        }
+
+        memoryStream.Position = 0;
+
+        var formFile = new FormFile(memoryStream, 0, memoryStream.Length, boreholeFile.File.Name, boreholeFile.File.Name)
+        {
+            Headers = new HeaderDictionary(),
+            ContentType = GetContentType(attachment.Name),
+        };
+        return formFile;
+    }
+
+    private static string GetContentType(string fileName)
+    {
+        var mimeType = MimeTypes.GetMimeType(Path.GetExtension(fileName));
+        return string.IsNullOrEmpty(mimeType) ? "application/octet-stream" : mimeType;
+    }
+
     private static bool ValidateFile(IFormFile file, Func<IFormFile, bool> fileValidationFunc)
     {
         if (file == null || file.Length == 0) return false;
         return fileValidationFunc(file);
     }
 
+    private void ValidateAttachmentsPresent(IEnumerable<string> attachmentsInZip, List<BoreholeImport> boreholesFromFile)
+    {
+        var referencedAttachments = boreholesFromFile
+            .Select(b => b.BoreholeFiles)
+            .Where(bf => bf != null)
+            .SelectMany(bf => bf!)
+            .Select(bf => bf.File.NameUuid + "_" + bf.File.Name);
```

Comment:
Q: Woher kommt dieser Aufbau? Kurz kommentieren, weshalb die Namen in dieser Form erwartet werden.